fw/apm.c

```diff
@@ -1652,6 +1713,12 @@ tfw_apm_cfgend(void)
 {
 	if (tfw_runstate_is_reconfig())
 		return 0;
+
+	tfw_apm_global_data = tfw_apm_data_create();
+	if (!tfw_apm_global_data)
+		return -EINVAL;
```

Comment:
-ENOMEM

Reply:
Fixed, but not only ENOMEM, see in further commit